src/components/KCollapse/KCollapse.vue

```diff
@@ -1,81 +1,79 @@
+        :is="titleTag"
         v-if="title"
-        class="k-collapse-title"
-        :class="{ 'has-trailing-trigger': trailingTrigger }"
-        data-testid="k-collapse-title"
+        class="collapse-title"
+        data-testid="collapse-title"
       >
         {{ title }}
-      </div>
-      <div
-        class="k-collapse-trigger"
-        :class="{ 'has-trailing-trigger': trailingTrigger }"
-      >
+      </component>
+      <div class="collapse-trigger">
         <slot
           :is-collapsed="collapsedState"
           name="trigger"
           :toggle="toggleDisplay"
         >
-          <a
-            class="k-collapse-trigger-content"
-            data-testid="k-collapse-trigger-content"
-            href="#"
+          <button
+            :aria-controls="contentId"
+            :aria-expanded="!collapsedState"
+            :aria-label="triggerLabel ? undefined : 'Toggle content'"
+            class="collapse-trigger-content"
+            data-testid="collapse-trigger-content"
             role="button"
             @click.prevent.stop="toggleDisplay()"
           >
             <slot name="trigger-content">
+              <ChevronRightIcon
+                class="collapse-trigger-icon"
+                :class="{ 'collapse-expanded': !collapsedState }"
+                data-testid="collapse-trigger-icon"
+                :size="KUI_ICON_SIZE_40"
+              />
               <span
                 v-if="triggerLabel"
-                class="k-collapse-trigger-label"
-                data-testid="k-collapse-trigger-label"
+                class="collapse-trigger-label"
+                data-testid="collapse-trigger-label"
               >
-                <KIcon
-                  class="k-collapse-trigger-chevron"
-                  :icon="collapseIcon"
-                  :size="KUI_ICON_SIZE_20"
-                />
-                <span>{{ triggerLabel }}</span>
+                {{ triggerLabel }}
               </span>
-              <KIcon
-                v-else
-                class="k-collapse-trigger-icon k-collapse-trigger-chevron"
-                data-testid="k-collapse-trigger-icon"
-                :icon="collapseIcon"
-              />
             </slot>
-          </a>
+          </button>
         </slot>
       </div>
     </div>
     <div
       v-if="hasVisibleContent"
-      class="k-collapse-visible-content"
-      data-testid="k-collapse-visible-content"
+      class="collapse-visible-content"
+      data-testid="collapse-visible-content"
     >
       <slot name="visible-content" />
     </div>
-    <div
-      v-show="!collapsedState"
-      class="k-collapse-hidden-content"
-      data-testid="k-collapse-hidden-content"
-    >
-      <slot />
-    </div>
+    <Transition name="kongponents-fade-transition">
+      <div
+        v-show="!collapsedState"
+        :id="contentId"
+        class="collapse-hidden-content"
+        data-testid="collapse-hidden-content"
+      >
+        <slot />
```

Comment:
nit, just so it's clear
```suggestion
        <slot name="default" />
```